components/proxy/src/main/java/com/hotels/styx/proxy/BackendServicesRouter.java

```diff
@@ -131,30 +131,24 @@ public void onChange(Registry.Changes<BackendService> changes) {
                     requestLoggingEnabled,
                     longFormat,
                     originStatsFactory,
-                    poolSettings.connectionExpirationSeconds());
+                    poolSettings.connectionExpirationSeconds(),
+                    environment.httpMessageFormatter());
 
             ConnectionPool.Factory connectionPoolFactory = new SimpleConnectionPoolFactory.Builder()
                     .connectionFactory(connectionFactory)
                     .connectionPoolSettings(backendService.connectionPoolConfig())
                     .metricRegistry(originsMetrics)
                     .build();
 
-            StyxHttpClient healthCheckClient = healthCheckClient(backendService);
-
-            OriginHealthStatusMonitor healthStatusMonitor = healthStatusMonitor(backendService, healthCheckClient);
-
-            StyxHostHttpClient.Factory hostClientFactory = (ConnectionPool connectionPool) -> {
-                StyxHeaderConfig headerConfig = environment.styxConfig().styxHeaderConfig();
-                return StyxHostHttpClient.create(connectionPool);
-            };
+            OriginHealthStatusMonitor healthStatusMonitor = healthStatusMonitor(backendService);
```

Comment:
Good refactoring here in this file. Thanks :-D

Reply:
np